bot/cogs/bot.py

```diff
@@ -78,32 +80,35 @@ def __init__(self, bot: AutoShardedBot):
             if time.time()-self.previous_format_times[msg.channel.id] > 300 or msg.channel.id == DEVTEST_CHANNEL:
                 if msg.content.count("\n") >= 3:
                     try:
-                        # Some users formatted multi-line code using `'sip("
-                        # This fixes that by treating them as nothing.
-                        if "`" in msg.content:
-                            msg.content = msg.content[:-1][1:]
-                        tree = ast.parse(msg.content)
+                        # Filtering valid Python codeblocks and exiting if a valid Python codeblock is found
+                        if re.match(msg.content, "```(python|py)\n((?:.*\n*)+)```", re.IGNORECASE):
```

Comment:
`re.match` is too strict for this. If there's a valid python codeblock anywhere in the message, we want to ignore it. Use `re.search` instead, since `re.match` only matches from the beginning of the line.